context-monitoring/src/main/java/de/atb/context/persistence/monitoring/MonitoringDataRepository.java

```diff
@@ -435,40 +462,30 @@ public synchronized ResultSet executeSparqlSelectQuery(final BusinessCase busine
         if (query.trim().length() == 0) {
             throw new IllegalArgumentException("Query may not be empty!");
         }
-        String finalQuery = prepareSparqlQuery(businessCase, query);
-        Dataset dataset = getDataSet(businessCase);
-        dataset.begin(ReadWrite.WRITE);
-        try {
-            Model model = dataset.getDefaultModel();
-            OntModel ontModel = ModelFactory.createOntologyModel(OntModelSpec.OWL_DL_MEM, model);
-            QueryExecution qexec = QueryExecutionFactory.create(finalQuery, ontModel);
-            try {
-                result = qexec.execSelect();
-            } catch (QueryException qe) {
-                logger.error(qe.getMessage(), qe);
-            }
-            dataset.commit();
-        } catch (Exception e) {
-            dataset.abort();
-        } finally {
-            dataset.end();
-        }
-        return result;
+
+        final String finalQuery = prepareSparqlQuery(query);
+        final Dataset dataset = getDataSet(businessCase);
+
+        return transactional(dataset, null, () -> {
+            final Model model = dataset.getDefaultModel();
+            final OntModel ontModel = ModelFactory.createOntologyModel(OntModelSpec.OWL_DL_MEM, model);
+            final QueryExecution qexec = QueryExecutionFactory.create(finalQuery, ontModel);
+            return qexec.execSelect();
+        });
     }
 
-    public static synchronized String prepareSparqlQuery(final BusinessCase businessCase, final String query) {
-        logger.trace("Preparing sparql query '" + query + "' for business case " + businessCase);
+    public static synchronized String prepareSparqlQuery(final String query) {
```

Comment:
This method and `executeSparqlSelectQuery` are very similar (if not identical) to methods `executeSparql`, `getQueryExecution`, and `prepareSparqlQuery` in `ContextRepository`.

Maybe they could be "merged" and moved to common base class `RepositoryTDB`?

Reply:
should be optimised now